### DISCUSSION TITLE
Development Setup

### DISCUSSION THREAD

**Discussion by joezimjs** [powerpaul17/nc_money#37] [Q&A] [answered]
I'm a Vue developer, and seeing this having so many little bugs, I naturally want to help fix this thing up and make it a great app, but I've never worked on any Nextcloud apps, so I'm curious what needs to be done to set up a development environment for this. Is there some documentation you can link to? Can it be worked on without embedding it into a Nextcloud environment?

**Comment by powerpaul17** [comment, marked as answer]
Hello and thanks for your interest in this app. First of all I must say, that I have still not published version 1 so there can (and probably will be) changes in UI, DB and pretty much all areas of the app and architecture. I'm still finding the best way for things to work. Unfortunately between work and family I have very limited time so progress is also slow.

Just out of curiosity: what "bugs" are you referring to?

Regarding development setup: in my opinion this is something Nextcloud is really missing at the moment: better documentation regarding development setup. I personally use VS Code and have setup a docker-compose file which creates a development stack with Nextcloud and MariaDB. The repository is then mounted inside the apps folder. While this works reasonably well it is still not completely the way I want it. Also many app developers seem to have similar approaches while they all seem to be specific for their workflow.

If you need general information about a Nextcloud app works, you can check out the documentation here: https://docs.nextcloud.com/server/stable/developer_manual

I was thinking about if it's possible to work on the frontend separately but you would have to mock the api calls which seems to be the same amount of work than setting it up in a container in my opinion. Also there would be all the framework and styles from nextcloud missing.

**Reply by joezimjs** [reply]
Sorry I haven't gotten to work on this at all. I was really hoping to at first because I was looking for a good way of tracking my expenses, but I'm learning more and more that unless I can have a system that pulls all transactions from my credit cards or something, I'll never keep it fully up-to-date, so this app really won't work well enough.
Thanks, though, for your work!

**Reply by powerpaul17** [reply]
Hey, that's a pity to hear but I totally understand. I really would like to have automatic fetching of card transactions but here in Europe it's still not possible in a way so it would work for every bank.. :slightly_frowning_face: I also have to import the transactions every month or so which makes it kind of difficult to get a continuous overview of my finances.. maybe there will be a solution sometime.

**Comment by joezimjs** [comment]
Thanks for this. I'll look more into when I get a chance, but like you, I'm a busy family man, so I don't foresee a lot of work getting done by me, but I'll do what I can. The bugs I'm seeing mostly involve showing transactions in the right pane from the previously selected account when creating a new account and needing to refresh the page to actually see updates to data.

**Reply by powerpaul17** [reply]
FYI, I also stumbled accross the issue with the transactions list not updating when creating a account. I've put it on my list and try to fix it soon.. :slightly_smiling_face:

**Reply by powerpaul17** [reply]
Fixed in 864cba15ebca5544b1a21867bec460bac4ec0236
